rewrite-maven/src/main/java/org/openrewrite/maven/RemoveDeadRepos.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.AllArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+import org.openrewrite.Recipe;
+
+public class RemoveDeadRepos extends Recipe {
+    private static final XPathMatcher REPOSITORY_URL_MATCHER = new XPathMatcher("/project/repositories/repository");
+    private static final XPathMatcher PLUGIN_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/pluginRepositories/pluginRepository");
+    private static final XPathMatcher DISTRIBUTION_MANAGEMENT_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/distributionManagement/repository");
+    private static final XPathMatcher DISTRIBUTION_MANAGEMENT_SNAPSHOT_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/distributionManagement/snapshotRepository");
+
+    @Override
+    public String getDisplayName() {
+        return "Remove dead repositories";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove repos marked as dead in Maven POM files.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RemoveDeadReposVisitor();
+    }
+
+
+    private static class RemoveDeadReposVisitor extends MavenIsoVisitor<ExecutionContext> {
+        @Override
+        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+            Xml.Tag newTag = super.visitTag(tag, ctx);
+            if (isRepositoryTag()) {
+                newTag = tag.getChild("url").flatMap(Xml.Tag::getValue).map(url -> {
```

Comment:
Reading this it looks like `newTag` will be a new instance of `Xml.Tag`, even if no changes should have been made.
Rewrite detects changes based on referential equality. I may be reading this incorrectly, but it looks like this would result in new objects being created even when no changes are intended.

Could you please have this use `ListUtils` to do any mapping operations you need. It was designed to be conscious of the importance of referential equality, and is consistent with the rest of our codebase. Since our OSS community learns how to write recipes in part from our repositories, consistency is useful for teaching as well.